src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.cs

```diff
@@ -149,5 +151,13 @@ internal void InitializeExtendedAttributesWithExisting(IEnumerable<KeyValuePair<
             TarEntryFormat.Gnu => GnuVersion,
             _ => string.Empty,
         };
+
+        // Stores the archive stream's position where we know the current entry's data section begins,
+        // if the archive stream is seekable. Otherwise, -1.
+        private static void SetDataOffset(TarHeader header, Stream archiveStream) => header._dataOffset =
+            archiveStream.CanSeek
+            // Add one because the last byte read is still part of the header
+            ? archiveStream.Position + 1
```

Comment:
Are we guaranteed to be given a `stream` with the `Position` starting at 0?  What if the stream has a bunch of other data and we're reading a tar from the middle of it?

Reply:
We discussed this in the API review too. The position of the offset would be absolute to the underlying archive stream, even if the user started reading the archive from the middle of the stream, the offset would be absolute to position zero of the archive stream anyway. I think this is the safest and least convoluted solution. Since the user is expected to own the underlying archive stream, they should be able to tell the offset relative to that archive stream.

I should mention this in the remarks.

Reply:
Ok, so long as it's documented clearly.  I can imagine some folks wanting this to represent the offset in the archive and not the stream.  Potentially in the future we could add another member which is offset in the archive.  In such a case it might be interesting (and expected) to handle the case Stephen was suggesting for non-seekable streams.